src/mappings/DAOTracker/mapping.ts

```diff
@@ -0,0 +1,121 @@
+import { Address, ipfs, store } from '@graphprotocol/graph-ts';
+import {
+  BlacklistDAO,
+  DAOTracker,
+  ResetDAO,
+  TrackDAO,
+} from '../../types/DAOTracker/DAOTracker';
+import {
+  AvatarContract,
+  BlacklistedDAO,
+  DAOTrackerContract,
+  ResetDAO as ResetDAOEntity,
+  UControllerOrganization,
+} from '../../types/schema';
+import {
+  Avatar_0_0_1_rc_31,
+  Controller_0_0_1_rc_31,
+  DAOToken_0_0_1_rc_31,
+  Reputation_0_0_1_rc_31,
+} from '../../types/templates';
+import { equalStrings } from '../../utils';
+
+export function getDAOTrackerContract(address: Address): DAOTrackerContract {
+  let daoTracker = DAOTrackerContract.load(address.toHex()) as DAOTrackerContract;
+  if (daoTracker == null) {
+    daoTracker = new DAOTrackerContract(address.toHex());
+    daoTracker.address = address;
+    let daoTrackerSC = DAOTracker.bind(address);
+    daoTracker.owner = daoTrackerSC.owner();
+    daoTracker.save();
+  }
+  return daoTracker;
+}
+
+export function handleTrackDAO(event: TrackDAO): void {
+  // Ensure the DAOTrackerContract has been added to the store
+  getDAOTrackerContract(event.address);
+
+  let avatar = event.params._avatar;
+  let controller = event.params._controller;
+  let reputation = event.params._reputation;
+  let daoToken = event.params._daoToken;
+
+  // If the avatar already exists, early out
+  if (AvatarContract.load(avatar.toHex()) != null) {
+    return;
+  }
+
+  // Tell the subgraph to start indexing events from the:
+  // Avatar, Controller, DAOToken, and Reputation contracts
+  Avatar_0_0_1_rc_31.create(avatar);
```

Comment:
how about : 
- create a template_create module which hold all templates history.
then the code should looks like:  
```
let contractInfo = ContractInfo.load(avatar.toHex());
  if (contractInfo != null) {
     tempate_create("avatar",contractInfo.version);
  }
```

Reply:
Oooo I really like this approach. Will look into how to implement this.

Reply:
Okay so this actually doesn't make sense @orenyodfat, since Avatar is a newly deployed Avatar that we don't know the address of at build-time. This means that it wouldn't be contained within ContractInfo.

We cannot know the ABI version of these newly added / tracked contracts for Avatar, Controller, DAOToken, and Reputation since we don't have factories that're spawning them to go off of.

Reply:
so we cannot be sure that the avatar is from v 0_0_1_rc_31 either. right ?
how about `TrackDAO` event will give us clue re arc version ?

- DAOCreator will be index and added to contractInfo.
- DAOTracker track will get arc version as a param : 
 `function track(Avatar _avatar, Controller _controller, string _arcVersion)`
- TrackDAO will emit the msg.sender and _arcVersion  along with all other addresses : 
`    event TrackDAO(address indexed _avatar, address _controller, address _reputation, address _daoToken, address _sender, string _arcVersion);
- if  _sender is DAOCreator (we can check that via contractInfo) so the avatar version is the DAOCreator version. - in this case the _arcVersion is ignored.
-  if  _sender is not DAOCreator the avatar version is _arcVersion.

Reply:
https://github.com/daostack/arc/pull/683

Reply:
https://github.com/daostack/migration/pull/219

Reply:
Should the DAOTracker mapping support both the old & new TrackDAO events?